pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -17,49 +17,44 @@
 
 import com.linkedin.pinot.core.io.compression.ChunkDecompressor;
 import com.linkedin.pinot.core.io.reader.BaseSingleColumnSingleValueReader;
-import com.linkedin.pinot.core.io.reader.impl.VarByteReaderContext;
-import com.linkedin.pinot.core.io.writer.SingleColumnSingleValueWriter;
+import com.linkedin.pinot.core.io.reader.impl.ChunkReaderContext;
 import com.linkedin.pinot.core.segment.memory.PinotDataBuffer;
 import java.io.IOException;
 import java.nio.ByteBuffer;
-import java.nio.charset.Charset;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 
 /**
- * Reader class for data written out by {@link SingleColumnSingleValueWriter}.
- * For data layout, please refer to the documentation for {@link SingleColumnSingleValueWriter}
+ * Abstract class implementation for {@link BaseSingleColumnSingleValueReader}.
+ * Base class for the fixed and variable byte reader implementations.
+ *
  */
-public class VarByteSingleValueReader extends BaseSingleColumnSingleValueReader<VarByteReaderContext> {
-  private static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
-  private static final Charset UTF_8 = Charset.forName("UTF-8");
+public abstract class BaseChunkSingleValueReader extends BaseSingleColumnSingleValueReader<ChunkReaderContext> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BaseChunkSingleValueReader.class);
+
+  protected static final int INT_SIZE = Integer.SIZE / Byte.SIZE;
+  protected static final int LONG_SIZE = Long.SIZE / Byte.SIZE;
+  protected static final int FLOAT_SIZE = Float.SIZE / Byte.SIZE;
+  protected static final int DOUBLE_SIZE = Double.SIZE / Byte.SIZE;
 
-  private final PinotDataBuffer _dataBuffer;
+  protected final PinotDataBuffer _dataBuffer;
   final PinotDataBuffer _header;
-  private final ChunkDecompressor _chunkDecompressor;
-  private final int _maxChunkSize;
-
-  private final int _numDocsPerChunk;
-  private final int _numChunks;
-  private final int _lengthOfLongestEntry;
-
-  // Thread local (reusable) byte[] to read bytes from data file.
-  private final ThreadLocal<byte[]> _reusableBytes = new ThreadLocal<byte[]>() {
-    @Override
-    protected byte[] initialValue() {
-      return new byte[_lengthOfLongestEntry];
-    }
-  };
+  protected final ChunkDecompressor _chunkDecompressor;
+  protected final int _chunkSize;
```

Comment:
Its computed based on values read in the header. Since the header is read in by the base class, I just kept it in the base class.